docs/concurrency.org

```diff
@@ -1,68 +1,251 @@
+
+Coroutines can be either stackful or stackless, we are not gonna dive deep into this concept since most of the time you are going to use stackful coroutines since they allow you to suspend from within a nested stackframe (the only strength of stackless coroutines: efficiency)
+
+Coroutines can also be symmetric or asymmetric:
 
 **** Symmetric coroutines
-- Their only control-transfer mechanism is: 
-1. explicitly passing control to another coroutine
+The only way to transfer the control flow is by explicitly passing control **to another coroutine**.
 
 **** Asymmetric coroutines (called asymmetric because the control-transfer can go both ways)
 - They have two control-transfer mechanisms:
 1. invoking another coroutine which is going to be the subcoroutine of the calling coroutine
 2. suspending itself and giving control back to the caller
 
 *** Green threads (userland threads)
-- preemptive multitasking (PAS VRAI selon appel !)
-- managed by a VM/runtime instead of the OS 
-- has the advantage to be managed in user space instead of kernel space which allows to avoid the overhead of making system calls to open and close threads
-- still use several native threads behind the scenes
-- used more for short-lived tasks
+Green threads, also known as userland threads are managed by a runtime or VM instead of the OS with standard kernel threads. They are lightweight compared to OS threads as they have a lower overhead (since it is managed in userspace instead of kernel). Even though they are not real OS threads, there are still OS threads that manage them under the hood. Paritculary useful for short-lived tasks.
+- preemptive multitasking ou cooperative (TODO JE NE SAIS TOUJOURS PAS psq le runtime peut toujours les preempter)
 
 *** Fibers
-- Same as green threads but cooperative multitasking instead of preemptive multitasking
+Same as green threads but cooperative multitasking instead of preemptive multitasking
 
 *** Preemptive multitasking
-- The underlying architecture (NOT us, but the OS for exemple) is going to be deciding what to execute and when
-- This means that our threads can be interrupted at any time
+In preemptive multitasking it is the underlying architecture (not us, but the OS or the runtime for exemple) that is in charge of choosing which threads to execute and when. This implies that our threads can be stopped (preempted) at any time, even if in the middle of a task. This method gives the advantage of not having to worr about a thread being starved, since the underlying architecture is going to make sure that everyone gets enough CPU time.
```

Comment:
typo error: `worr`

Reply:
and starved ?

Reply:
yes, by that I mean that it not possible that a thread is in "famine" and is never runned

Reply:
fixed the typo btw